tools/c7n_azure/c7n_azure/resources/key_vault.py

```diff
@@ -114,3 +119,48 @@ def enhance_policies(self, access_policies):
             policy['principalName'] = GraphHelper.get_principal_name(aad_object)
 
         return access_policies
+
+
+@resources.register('key')
+class KeyVaultKey(ArmResourceManager):
+
+    class resource_type(ArmResourceManager.resource_type):
+        service = 'azure.mgmt.keyvault'
+        client = 'KeyVaultManagementClient'
+        enum_spec = ('vaults', 'list', None)
+
+    def augment(self, resources):
+        url_base = 'vault.azure.net'
+        s = Session(resource='https://' + url_base)
+        self.keyvault_client = KeyVaultClient(s.get_credentials())
+        r = []
+        for index, item in enumerate(resources):
```

Comment:
You can do for item in resources:

and then use item instead of resources[index]

Reply:
Done, will be in next commit.